### ISSUE TITLE
last retropie stretch error 

### ISSUE THREAD

**Issue by pietrondo** [ArthurMoore85/pi_romulus#22]
I'm looking for some dreamcast roms and after, when i select the roms before the start of the download:

`Traceback (most recent call last):
  File "romulus.py", line 34, in <module>
    app.run()
  File "/usr/local/lib/python2.7/dist-packages/npyscreen/apNPSApplication.py", line 30, in run
    return npyssafewrapper.wrapper(self.__remove_argument_call_main)
  File "/usr/local/lib/python2.7/dist-packages/npyscreen/npyssafewrapper.py", line 41, in wrapper
    wrapper_no_fork(call_function)
  File "/usr/local/lib/python2.7/dist-packages/npyscreen/npyssafewrapper.py", line 97, in wrapper_no_fork
    return_code = call_function(_SCREEN)    
  File "/usr/local/lib/python2.7/dist-packages/npyscreen/apNPSApplication.py", line 25, in __remove_argument_call_main
    return self.main()
  File "/usr/local/lib/python2.7/dist-packages/npyscreen/apNPSApplicationManaged.py", line 174, in main
    self._THISFORM.afterEditing()
  File "/home/pi/RetroPie/pi_romulus/forms/results.py", line 33, in afterEditing
    self.emu.download(selection)
  File "/home/pi/RetroPie/pi_romulus/api/base.py", line 151, in download
    self.current_url = self.get_download_url()
  File "/home/pi/RetroPie/pi_romulus/api/providers/emuapi.py", line 73, in get_download_url
    link = self.base_url + soup.find('a', {'id': 'download-link'}).get('href')
AttributeError: 'NoneType' object has no attribute 'get'
`

python version:
Python 2.7.13

debian stretch

thx

**Comment by ArthurMoore85** [comment]
Thank you for reporting this issue. I will investigate what's going on there and try and get it fixed as soon as possible.

Regards,
Arthur

**Comment by ArthurMoore85** [comment]
This issue is now fixed and merged into master.
However, I did encounter an issue with the extraction of the 7z files it downloads.
That bug is out of my hands as it is caused by Pylzma. The bug has already been raised but thusfar has not been fixed.
I will keep this ticket on hold until that is fixed and the ticket is truly closed.

Dreamcast games can now be downloaded, but not extracted, so that will need to be done manually for now.
